src/redux/actions/loginAction.js

```diff
@@ -0,0 +1,48 @@
+import axios from "axios";
+import $ from "jquery";
+import history from "../../routes/history";
+import {
+  LOGIN_SUCCESSFUL,
+  LOGIN_ERROR,
+  LOGOUT_SUCCESSFUL
+} from "../action_types";
+import homepage from "../../views/Home/index";
+
+const url = "https://ah-magnificent6-staging.herokuapp.com/api/";
+
+export const loginSuccessful = message => ({
+  type: LOGIN_SUCCESSFUL,
+  message
+});
+
+export const loginError = error => ({
+  type: LOGIN_ERROR,
+  error
+});
+
+export const logoutsuccessful = () => ({
+  type: LOGOUT_SUCCESSFUL
+});
+
+export default user => dispatch => {
+  axios
+    .post(`${url}users/login/`, user)
+    .then(response => {
+      const message = response.data;
+      dispatch(loginSuccessful(message));
+      $(".close").click();
+      localStorage.setItem("token", message.response.token);
+      history.push(`${homepage}`);
+    })
+
```

Comment:
Why this extra empty line 37?

Reply:
@kimotho-njoki Which line? I seem not to be seeing an empty line.

Reply:
@kimotho-njoki We have seen and Resolved